rollup.config.js

```diff
@@ -1,3 +1,22 @@
+/**
+ * Custom Rollup plugin for `import.meta.url` transformation to commonjs.
+ * The default transformation ('file:' + __filename) does not check characters in __filename,
+ * and thus can produce invalid URLs, like in https://github.com/eslint/eslint/issues/15766
+ * See https://github.com/eslint/eslint/issues/15766#issuecomment-1093941321
+ * @returns {Object} Rollup plugin object.
+ */
+function importMetaURLPlugin() {
+    return {
+        name: "custom-import-meta-url",
+        resolveImportMeta(property) {
+            if (property === "url") {
+                return "require('url').pathToFileURL(__filename)";
```

Comment:
Just a note: `pathToFileURL` returns a URL object, while `import.meta.url` is a string. This is not a problem as long as `import.meta.url` is only used for `createRequire` as it is here the case. But if we accept that, then we could also just return `__filename`, because `createRequire` is perfectly fine with a path as argument.